### PR TITLE
Initial move from vertx-hawkular-metrics to global vertx-monitoring

### PR DESCRIPTION
**Pull request by jotak**
Based on PR https://github.com/vert-x3/vertx-hawkular-metrics/pull/65 from @tsegismont and @dankristensen

Includes only influxdb reporter at the moment

### CODE REVIEW

File: batching-collector/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  <parent>
+    <groupId>io.vertx</groupId>
+    <artifactId>vertx-monitoring-parent</artifactId>
+    <version>3.5.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>vertx-monitoring-batching-collector</artifactId>
+
+  <name>Vert.x Metrics Collector</name>
```
Comment (jotak) [inline batching-collector/pom.xml]: Vert.x Monitoring Batching Collector

File: batching-collector/src/main/java/io/vertx/ext/monitoring/collector/BatchingReporterOptions.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  /**
+   * @return the maximum number of metrics in a batch
+   */
+  public int getBatchSize() {
+    return batchSize;
+  }
+
+  /**
+   * Set the maximum number of metrics in a batch. To reduce the number of HTTP exchanges, metric data is sent to the
+   * Hawkular server in batches. A batch is sent as soon as the number of metrics collected reaches the configured
```
Comment (jotak) [inline batching-collector/src/main/java/io/vertx/ext/monitoring/collector/BatchingReporterOptions.java]: not Hawkular

File: batching-collector/src/main/java/io/vertx/ext/monitoring/collector/BatchingReporterOptions.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  /**
+   * @return the maximum delay between two consecutive batches
+   */
+  public int getBatchDelay() {
+    return batchDelay;
+  }
+
+  /**
+   * Set the maximum delay between two consecutive batches (in seconds). To reduce the number of HTTP exchanges, metric
+   * data is sent to the Hawkular server in batches. A batch is sent as soon as the number of metrics collected reaches
```
Comment (jotak) [inline batching-collector/src/main/java/io/vertx/ext/monitoring/collector/BatchingReporterOptions.java]: Not Hawkular

File: monitoring-base/src/main/asciidoc/enums.adoc [hunk 4]
```diff
@@ -0,0 +1,48 @@
+= Enums
+
+[[MetricsType]]
```
Comment (jotak) [inline monitoring-base/src/main/asciidoc/enums.adoc]: Maybe rename type => category? Metric type is often referred as things like gauge/counter/...

File: monitoring-base/src/main/java/io/vertx/ext/monitoring/base/MetricsOptionsBase.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+   */
+  @Override
+  public MetricsOptionsBase setEnabled(boolean enable) {
+    super.setEnabled(enable);
+    return this;
+  }
+
+  /**
+   * @return the metric bridge address. If enabled the metric bridge transfers metrics collected from the event bus to
+   * the Hawkular server. The metrics are sent as message on the event bus to the return address. The message is a
```
Comment (jotak) [inline monitoring-base/src/main/java/io/vertx/ext/monitoring/base/MetricsOptionsBase.java]: not Hawkular

File: reporters/influxdb/README.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
+= Vert.x InfluxDB Metrics
+:source-language: java
+
+image::https://vertx.ci.cloudbees.com/buildStatus/icon?job=vert.x3-hawkular-metrics[link=https://vertx.ci.cloudbees.com/job/vert.x3-hawkular-metrics/]
```
Comment (jotak) [inline reporters/influxdb/README.adoc]: fix urls

File: reporters/influxdb/README.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
+= Vert.x InfluxDB Metrics
```
Comment (jotak) [inline reporters/influxdb/README.adoc]: Rewrite readme. It's not really anymore about the SPI, it's a reporter implementation.

File: reporters/influxdb/src/test/java/io/vertx/ext/influxdb/impl/InfluxDbReporterTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        Buffer fullRequestBody = Buffer.buffer();
+        req.handler(fullRequestBody::appendBuffer);
+        req.endHandler(h -> {
+          context.assertTrue(fullRequestBody.toString().contains("eventbus.publishedRemoteMessages=0i"));
+          req.response().setStatusCode(200).end();
+          async.complete();
+        });
+      }).listen(8086, "localhost", context.asyncAssertSuccess());
+    async.awaitSuccess();
+  }
```
Comment (jotak) [inline reporters/influxdb/src/test/java/io/vertx/ext/influxdb/impl/InfluxDbReporterTest.java]: There was more tests under the hawkular reporter. They must be added back here and/or under the collector package.

**Comment by jotak** [comment]
@vietj can you create the jenkins job for vertx-monitoring? (the readme.adoc build badge expects url: https://vertx.ci.cloudbees.com/job/vert.x3-monitoring/ )

**Comment by vietj** [comment]
I can add you as admin on Jenkins if you give me your cloudbees ID

**Comment by jotak** [comment]
@vietj ok, there's none at the moment, let me create an account

**Comment by jotak** [comment]
Closing, since the commits have been moved to branch `initial-work` and replaced with #9 
(with jenkins build passing after several patches)

**Pull request closed without merging**